data_transfer/services/ucam.py

```diff
@@ -65,6 +65,42 @@ def __device_from_record(device: Payload) -> Device:
     return PatientRecord(patient, devices)
 
 
+def record_by_vtt(vtt_hash: str) -> Optional[PatientRecord]:
+    """
+    Return a patient record based on then Hashed ID provided by VTT
+    The VTT hashes are unique to each patient
+
+    GET /vtt/<vtt_hash>/      
+    """
+    patient_records = [r for r in __get_patients() if r.vttsma_id == vtt_hash]
+    
+    # No records exist for that patient, 
+    # e.g., if a device was not worn or a staff member forget to add the record
+    if len(patient_records) == 0:
+        return None
+
+    # Records returned from UCAM contain the patient ID in each row.
+    # Rather than duplicating this, we create it once, then associate
+    # all other rows (i.e., devices) below
+    patient = patient_records[0]
+    patient = Patient(patient.patient_id, patient.disease)
+
+    def __device_from_record(device: Payload) -> Device:
+        """
+        Convenient method to only store Device-specific metadata.
+        """
+        return Device(
+            id=device.device_id, 
+            vttsma_id=device.vttsma_id, 
+            devitations=device.devitations,
+            start_wear=device.start_wear,
+            end_wear=device.end_wear
+        )
+
+    devices = [__device_from_record(r) for r in patient_records]
+    return PatientRecord(patient, devices)
+
+
```

Comment:
Maybe refactor this to a new method (__serialise_records) and call it here and in `get_record` since the code is 1-1?

Reply:
Good point. Done and tested in this [commit](https://github.com/ideafast/middleware-services/pull/21/commits/cb462bc04765804a55acbfd9a5605f72315a2c96).